src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotDirectlyAwaitATask.cs

```diff
@@ -96,6 +101,70 @@ private static void AnalyzeAwaitOperation(OperationAnalysisContext context, Immu
             }
         }
 
+        private static void AnalyzeForEachOperation(OperationAnalysisContext context,
+            INamedTypeSymbol configuredCancelableAsyncEnumerable)
+        {
+            if (context.Operation is not IForEachLoopOperation operation)
+                return;
+
+            if (!operation.IsAsynchronous())
+            {
+                return;
+            }
+
+            // Get the type of the expression being iterated over.
+            IOperation collectionOperation = operation.Collection;
+            ITypeSymbol? typeOfCollection = collectionOperation.Type;
+            if (typeOfCollection.OriginalDefinition == configuredCancelableAsyncEnumerable)
+            {
+                // Operation should be conversion to IAsyncEnumerable
+                if (collectionOperation is IConversionOperation conversionOperation)
+                {
+                    if (conversionOperation.Operand is ILocalReferenceOperation localReferenceOperation)
+                    {
+                        // Can we find the initializer of this local?
+                        var location = localReferenceOperation.Local.Locations.First();
+                        SyntaxNode? node = location.SourceTree?.GetRoot(context.CancellationToken)
+                                                               .FindNode(location.SourceSpan);
+                        if (node is not null)
+                        {
+                            // Checking VariableDeclaratorSyntax.Initializer is CSharp specific
+                            // Do we need to move rule to CSharp.NetAnalyzers?
+                            if (node.ToString().Contains(ConfigureAwait))
+                            {
+                                return;
+                            }
+                        }
```

Comment:
💭 This check should be possible using the operation tree directly.

Reply:
Yes, found a way to obtain `IVariableDeclaratorOperation` and from there the `IVariableInitializerOperation`.